class/frontend-account.php

```diff
@@ -160,28 +160,16 @@ public function shortcode( $atts ) {
         if ( is_user_logged_in() ) {
             $default_active_tab = wpuf_get_option( 'account_page_active_tab', 'wpuf_my_account', 'dashboard' );
             $section            = isset( $_REQUEST['section'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['section'] ) ) : $default_active_tab;
-            $allowed_posts      = $this->get_allowed_cpt();
-            if ( '' !== $allowed_posts ) {
-                add_filter(
-                    'wpuf_account_sections', function ( $account_sections ) use ( $allowed_posts ) {
-                        foreach ( $allowed_posts as $post_type ) {
-                            $cpt['slug']  = $post_type;
-                            $cpt['label'] = ucfirst( $post_type ) . 's';
-                            array_splice( $account_sections, 1, 0, [ $cpt ] );
-                        }
-                        return $account_sections;
-                    }
-                );
-            } else {
-                add_filter(
-                    'wpuf_account_sections', function ( $account_sections ) {
-                        $post['slug']  = 'post';
-                        $post['label'] = __( 'Posts', 'wp-user-frontend' );
-                        array_splice( $account_sections, 1, 0, [ $post ] );
-                        return $account_sections;
+
+            add_filter(
+                'wpuf_account_sections', function ( $sections ) {
+                    $sections_new = [];
```

Comment:
@zamandevs why do we still need this one? We can run the loop without using the `add_filter`, then pass the new mapped sections to the template.

As the `wpuf_get_account_sections()` function call is on the top, anyone will be able to change the labels with filter.

Reply:
@tareq1988 Current section is indexed numerically. we remap this index with section slug thus user can change slug without checking the current slug. if we want to keep sections like this we need a filter before/after calling the section function. I am using the previous filter without introducing a new one before calling the function. or we need to add another filter after the calling function if we want to index section slug as key. 

if we remap user can change `add_filter('wpuf_account_sections',function ($sections){ $sections['post']['label'] = 'List'; $sections['subscription']['label'] = 'Membership'; return $sections; },999);` like this otherwise he must check current slug.

Reply:
Then can we change our main function `wpuf_get_account_sections()` like this, so we won't have to worry about indexes.

```php
$default_sections = [
    'dashboard'       => __( 'Dashboard', 'wp-user-frontend' ),
    'edit-profile'    => __( 'Edit Profile', 'wp-user-frontend' ),
    'subscription'    => __( 'Subscription', 'wp-user-frontend' ),
    'billing-address' => __( 'Billing Address', 'wp-user-frontend' ),
];
```

We can keep it backward compatible by passing it to the template like this:

```php
$default_sections = wpuf_get_account_sections();
$sections = [];

// format the sections
foreach ( $default_sections as $key => $label ) {
    // check backward compatibility
    if ( is_array( $label ) ) {
        $sections[ $label['slug'] ] = $label['label']
    } else {
        $sections[ $key ] = $label;
    }
}
```

Reply:
Firstly thought like this, but this function was called other places which might break current implementation thus i've remap index here.